src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -86,20 +131,45 @@ private static bool MatchesDirective(TagHelperDescriptor descriptor, string type
             return string.Equals(descriptor.Name, typePattern, StringComparison.Ordinal);
         }
 
-        internal class DirectiveVisitor : SyntaxRewriter
+        private static bool IsComponentTagHelperKind(string tagHelperKind)
+        {
+            return tagHelperKind == ComponentMetadata.Component.TagHelperKind ||
+                tagHelperKind == ComponentMetadata.ChildContent.TagHelperKind ||
+                tagHelperKind == ComponentMetadata.EventHandler.TagHelperKind ||
+                tagHelperKind == ComponentMetadata.Bind.TagHelperKind ||
+                tagHelperKind == ComponentMetadata.Ref.TagHelperKind;
+        }
```

Comment:
Can we make this an extension method, or put this somewhere more obvious? We will probably add more things in the future.